### PR TITLE
store.updateOrAdd

### PR DESCRIPTION
**Pull request by gr2m**
part of #2

### CODE REVIEW

File: lib/update-or-add.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+module.exports = function updateOrAdd (idOrObjectOrArray, newObject) {
+  var state = {
+    db: this.db,
+    Promise: this.PouchDB.utils.Promise,
+    errors: this.PouchDB.Errors
+  }
+  var isArray = Array.isArray(idOrObjectOrArray)
+
+  return isArray ? updateOrAddMany(state, idOrObjectOrArray) : updateOrAddOne(state, idOrObjectOrArray, newObject)
```
Comment (adamyeats-zz) [inline lib/update-or-add.js:14]: I like this :+1:

**Comment by adamyeats-zz** [comment]
:+1: looks good to me!

**Comment by boennemann** [comment]
LGTM

**Comment by gr2m** [comment]
@boennemann I've left out the exporting of the internal utils, as we don't need it for testing atm. I'd add it later if needed. Code is cleaner without.

**Pull request merged**